apps/ai-image-generator/backend/src/actions/aiig-generate-image.ts

```diff
@@ -0,0 +1,41 @@
+import { AppActionCallContext } from '@contentful/node-apps-toolkit';
+import { PlainClientAPI } from 'contentful-management';
+import { OpenAiApiService } from '../services/openaiApiService';
+
+interface AppActionCallParameters {
+  prompt: string;
+}
+
+interface AppActionResponse {
+  status: number;
+  prompt: string;
+  images: string[];
+}
+
+async function fetchOpenAiApiKey(_cma: PlainClientAPI) {
+  return 'openai-api-key';
+}
+
+async function makeOpenAiApiService(cma: PlainClientAPI): Promise<OpenAiApiService> {
+  const openAiApiKey = await fetchOpenAiApiKey(cma);
+  return OpenAiApiService.fromOpenAiApiKey(openAiApiKey);
+}
+
+export const handler = async (
+  payload: AppActionCallParameters,
+  context: AppActionCallContext,
+  openAiApiService?: OpenAiApiService
```

Comment:
Yeah, this is basically dependency injection for test purposes. Unfortunately using sinon.stub was not actually working. It works fine in our other apps, which was super confusing. 😢

Reply:
Okay weird but I think this is good because the comment I originally made was going to be centered around ```makeOpenAiApiService```.

I personally think this should not even exist as a function. We should just pull the API key, and then all the services should only take releveant parms like the API key, prompt etc...

This way the class OpenAiApiService object should have a class variable that holds the API Key.
```
export const handler = async (
  payload: AppActionCallParameters,
  context: AppActionCallContext,
  openAiApiService?: OpenAiApiService
): Promise<AppActionResponse> => {
  // TODO: Check input params

  const apiKey = fetchOpenAiApiKey(context.cma)
  const openAiApiService = new OpenAiApiService(apiKey)
  const { prompt } = payload;
  const images = await openAiApiService.createImage({
```

I personally think this makes things cleaner, is more clear, and avoids this whole service injection for testing thing

Reply:
However, I think this is a side issue as this PR is about getting the app action up and getting it to call openAI apis. So I'll approve unless you disagree otherwise, and you/I can fix optimize this part with the suggestion above